src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVECandidates.hpp

```diff
@@ -0,0 +1,190 @@
+
+            if (affected->versions())
+            {
+                for (const auto& versionElement : *affected->versions())
+                {
+                    NSVulnerabilityScanner::Status status;
+                    flatbuffers::Offset<flatbuffers::String> version;
+                    flatbuffers::Offset<flatbuffers::String> lessThan;
+                    flatbuffers::Offset<flatbuffers::String> lessThanOrEqual;
+                    flatbuffers::Offset<flatbuffers::String> versionType;
+
+                    if (versionElement->status())
+                    {
+                        status = VERSION_STATUS_MAP.at(versionElement->status()->str());
+                    }
+
+                    if (versionElement->version())
+                    {
+                        version = builder.CreateString(versionElement->version()->str());
+                    }
+
+                    if (versionElement->lessThan())
+                    {
+                        lessThan = builder.CreateString(versionElement->lessThan()->str());
+                    }
+
+                    if (versionElement->lessThanOrEqual())
+                    {
+                        lessThanOrEqual = builder.CreateString(versionElement->lessThanOrEqual()->str());
+                    }
+
+                    if (versionElement->versionType())
+                    {
+                        versionType = builder.CreateString(versionElement->versionType()->str());
+                    }
+
+                    NSVulnerabilityScanner::VersionBuilder versionBuilder(builder);
+                    versionBuilder.add_status(status);
+                    versionBuilder.add_version(version);
+                    versionBuilder.add_lessThan(lessThan);
+                    versionBuilder.add_lessThanOrEqual(lessThanOrEqual);
+                    versionBuilder.add_versionType(versionType);
+
+                    versionFBArray.push_back(versionBuilder.Finish());
+                }
+            }
+            auto versionsFB = builder.CreateVector(versionFBArray);
+
+            std::vector<flatbuffers::Offset<flatbuffers::String>> platformsVec;
+
+            if (affected->platforms())
+            {
+                for (const auto& platform : *affected->platforms())
+                {
+                    platformsVec.push_back(builder.CreateString(platform->str()));
+                }
+            }
+
+            auto platformsFB = builder.CreateVector(platformsVec);
+
+            NSVulnerabilityScanner::ScanVulnerabilityCandidateBuilder scanVulnerabilityCandidateBuilder(builder);
+
+            scanVulnerabilityCandidateBuilder.add_cveId(cveIdFB);
+            scanVulnerabilityCandidateBuilder.add_versions(versionsFB);
+
+            if (affected->defaultStatus())
+            {
+                scanVulnerabilityCandidateBuilder.add_defaultStatus(
+                    VERSION_STATUS_MAP.at(affected->defaultStatus()->str()));
```

Comment:
What happens if the element status is not present in the VERSION_STATUS_MAP?

Reply:
Now the access to the map is safer